src/sentry/data_export/tasks.py

```diff
@@ -65,26 +71,46 @@ def assemble_download(
         return
 
     try:
+        # ensure that the export limit is set and capped at EXPORTED_ROWS_LIMIT
         if export_limit is None:
             export_limit = EXPORTED_ROWS_LIMIT
         else:
             export_limit = min(export_limit, EXPORTED_ROWS_LIMIT)
 
-        # if there is an export limit, the last batch should only return up to the export limit
-        if export_limit is not None:
-            batch_size = min(batch_size, max(export_limit - offset, 0))
-
         processor = get_processor(data_export, environment_id)
 
         with tempfile.TemporaryFile() as tf:
             writer = csv.DictWriter(tf, processor.header_fields, extrasaction="ignore")
             if first_page:
                 writer.writeheader()
 
-            rows = process_rows(processor, data_export, batch_size, offset)
-            writer.writerows(rows)
+            # the position in the file at the end of the headers
+            starting_pos = tf.tell()
+
+            # the row offset relative to the start of the current task
+            # this offset tells you the number of rows written during this batch
+            batch_offset = 0
+
+            # the absolute row offset from the beginning of the export
+            next_offset = offset + batch_offset
```

Comment:
Does this addition do anything as you are adding 0 all the time?

Reply:
Doesn't do anything code wise, but just wanted to be explicit that `next_offset` should be the sum of the two as this is used again later on.